kyo-core/shared/src/main/scala/kyo/Clock.scala

```diff
@@ -4,12 +4,60 @@ import java.time.Instant
 
 abstract class Clock:
     def now(using Frame): Instant < IO
+    def stopwatch(using Frame): Clock.Stopwatch < IO
+    def deadline(duration: Duration)(using Frame): Clock.Deadline < IO
+    def unsafe: Clock.Unsafe
+end Clock
 
 object Clock:
 
-    val live: Clock =
-        new Clock:
-            def now(using Frame) = IO(Instant.now())
+    abstract class Unsafe:
+        def now: Instant
+        def stopwatch: Unsafe.Stopwatch
+        def deadline(duration: Duration): Unsafe.Deadline
+    end Unsafe
+
+    object Unsafe:
+        class Stopwatch(start: Instant, clock: Clock.Unsafe):
+            def elapsed: Duration =
+                Duration.fromJava(java.time.Duration.between(start, clock.now))
+
+        class Deadline(end: Instant, clock: Clock.Unsafe):
+            def timeLeft: Duration =
+                val remaining = java.time.Duration.between(clock.now, end)
+                if remaining.isNegative then Duration.Zero else Duration.fromJava(remaining)
```

Comment:
I think we should eventually have an optimized version of `Instance` that integrates well with `Duration`